### PR TITLE
DOC: Fix and extend the docstring for np.inner

### PR DESCRIPTION
**Pull request by mwtarnowski**
This PR fixes a mistake in the description of `np.inner` function:
> More generally, if _ndim(a) = r > 0_ and _ndim(b) = s > 0_:
> ...
> ```
> np.inner(a, b)[i0,...,ir-1,j0,...,js-1]
>      = sum(a[i0,...,ir-1,:]*b[j0,...,js-1,:])
> ```
In this case, the result of `np.inner(a,b)` is a `r+s-2`-dimensional array and indexing it with `r+s` indices will raise the `IndexError`.

For further clarification I have expanded the description and added one more example.

### CODE REVIEW

File: numpy/core/multiarray.py [hunk 1]
```diff
@@ -300,14 +304,25 @@ def inner(a, b):
     >>> np.inner(a, b)
     2
 
-    A multidimensional example:
+    A multidimensional examples:
```
Comment (eric-wieser) [inline numpy/core/multiarray.py:307]:
```suggestion
    Some multidimensional examples:
```

File: numpy/core/multiarray.py [hunk 2]
```diff
@@ -259,12 +259,16 @@ def inner(a, b):
     Returns
     -------
     out : ndarray
-        `out.shape = a.shape[:-1] + b.shape[:-1]`
+        Returns the inner product of `a` and `b`. If `a` and `b` are both
```
Comment (mattip) [inline numpy/core/multiarray.py:262]: Small cleanup

```suggestion
        If `a` and `b` are both
```

File: numpy/core/multiarray.py [hunk 3]
```diff
@@ -259,12 +259,16 @@ def inner(a, b):
...
+        scalars or both 1-D arrays then a scalar is returned; otherwise
+        an array is returned.
+        `out.shape = (*a.shape[:-1], *b.shape[:-1])`
```
Comment (mattip) [inline numpy/core/multiarray.py:265]: Mistake in the original code, might as well fix it here
```suggestion
        ``out.shape = (*a.shape[:-1], *b.shape[:-1])``
```

**Review by mattip — APPROVED** [review]
Thanks! A few small cleanups would be nice

**Comment by mattip** [comment]
Thanks @mwtarnowski

**Pull request merged**